modules/nextflow/src/main/groovy/nextflow/trace/PreviewReportWriter.groovy

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright 2013-2023, Seqera Labs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package nextflow.trace
+
+import java.nio.file.Path
+
+import groovy.json.JsonBuilder
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+import nextflow.Session
+import nextflow.dag.DAG
+import nextflow.exception.AbortOperationException
+import nextflow.file.FileHelper
+import nextflow.util.Duration
+import nextflow.util.MemoryUnit
+/**
+ * Render the preview report when running a pipeline
+ * in preview mode.
+ *
+ * @author Ben Sherman <bentshermann@gmail.com>
+ */
+@Slf4j
+@CompileStatic
+class PreviewReportWriter implements TraceObserver {
+
+    static private final List<String> DEF_DIRECTIVES = ['container', 'cpus', 'memory', 'time']
+
+    static private final String DEF_FILE_NAME = "preview-${TraceHelper.launchTimestampFmt()}.json"
+
+    private DAG dag
+
+    private List<String> directives
+
+    private Path file
+
+    private boolean overwrite
+
+    static PreviewReportWriter create(Session session) {
+        final directives = session.config.navigate('preview.directives', DEF_DIRECTIVES) as List
+        def file = session.config.navigate('preview.file', DEF_FILE_NAME)
+        file = (file as Path).complete()
+        final overwrite = session.config.navigate('preview.overwrite', false) as boolean
+
+        return new PreviewReportWriter(session.dag, directives, file, overwrite)
+    }
```

Comment:
would not be better to keep the creation login the same as for the other observers? 

https://github.com/nextflow-io/nextflow/blob/main/modules/nextflow/src/main/groovy/nextflow/trace/DefaultObserverFactory.groovy#L12-L12